### ISSUE TITLE
isLoading prop not updating in Toolbar

### ISSUE THREAD

**Issue by Firgrep** [mdx-editor/editor#94]
Hey,

I've added a "Save button" to the Toolbar, and it has a function passed down to it from the parent function that renders the MDXEditor. When I click the button, it triggers the function and everything works as expected. I wanted to add a disabled state while the mutation is running, however, it doesn't appear that the `isLoading` prop I pass down to the toolbar is being updated. Is there an intermediary that needs to be put in or something? Here is my code below. Appreciate any help! Cheers

Main Editor component that handles mutation logic and renders the MDXEditor (excerpt):

```ts
export default function Editor({ initialMaterial, title }: EditorProps) {
    const editorRef = React.useRef<MDXEditorMethods>(null);
    const updateMaterialMutation = //some api call
    console.log(updateMaterialMutation.isLoading) // this updates as expected
    ...
    return (
        <>
            <ForwardedRefMDXEditor 
                ref={editorRef}
                markdown={incomingMarkdown}
                contentEditableClassName="prose max-w-none"
                plugins={[
                    listsPlugin(),
                    quotePlugin(),
                    headingsPlugin(),
                    linkPlugin(),
                    linkDialogPlugin(),
                    imagePlugin(),
                    tablePlugin(),
                    thematicBreakPlugin(),
                    frontmatterPlugin(),
                    codeBlockPlugin({ defaultCodeBlockLanguage: 'txt' }),
                    codeMirrorPlugin({ codeBlockLanguages: { js: 'JavaScript', css: 'CSS', txt: 'text', tsx: 'TypeScript' } }),
                    directivesPlugin({ directiveDescriptors: [ AdmonitionDirectiveDescriptor] }),
                    diffSourcePlugin({ viewMode: 'rich-text', diffMarkdown: 'boo' }),
                    markdownShortcutPlugin(),
                    toolbarPlugin({
                        toolbarContents: () => (
                            <DefaultToolbar handleSave={handleSave} isLoading={updateMaterialMutation.isLoading} />
                        )
                    })
                ]}
    ...
```

Toolbar component (excerpt):
```ts
type DefaultToolbarProps = {
    handleSave: () => Promise<void>;
    isLoading: boolean;
}
const DefaultToolbar: React.FC<DefaultToolbarProps> = ({ handleSave, isLoading }) => {
    
    const handleSaveButton = async () => {
        await handleSave();
    }

    console.log("LOADING STATE: ", isLoading) // ! always remains false even when isLoading is true in the parent component

    return (
        <DiffSourceToggleWrapper>
            <ConditionalContents
                options={[
                    { when: (editor) => editor?.editorType === 'codeblock', contents: () => <ChangeCodeMirrorLanguage /> },
                    { when: (editor) => editor?.editorType === 'sandpack', contents: () => <ShowSandpackInfo /> },
                    { fallback: () => (
                        <>  
                            <TooltipWrap title="Save to database">
                                { isLoading
                                    ? <LoadingBars size="xs" />
                                    : <Button onClick={() => handleSaveButton()}>💾</Button>
                                }
                            </TooltipWrap>

    ...
```

**Comment by petyosi** [comment]
Yeah, the toolbar contents are not updating with each re-render, so the value is stale. You can use context provider/ `useContext` to transfer values from your state into the toolbar component, that should work as expected.

**Comment by Firgrep** [comment]
Alright, thank you for the superquick response!

I wrapped the `<ForwardedRefMDXEditor />` around a `<EditorContext.Provider value={updateMaterialMutation.isLoading} />` and then made used `useContext(EditorContext)` inside of the `<DefaultToolbar/>` and that seems to have done the trick! For anyone wondering, I just created the context inside the same file .tsx as it's just holding one value currently.
